pitch.md

```diff
@@ -23,38 +31,79 @@ should be more specific than "The user describes properties of unicorns.", e.g.,
+that experience was significantly small. 
+There may be a few different versions of this "generate" method that generates
+coloring pages with a different modifications. 
+I imagine that the syntax is not going to be that complicated because the main
+purpose of this DSL is to convert an image to a coloring page. 
+
 
 ## Operation (semantics)
 _What might happen when a program runs? Is there a computational model that
 corresponds with your domain? How does a program interact with the
 user? What kinds of semantic (i.e., non-syntax) errors might occur, and how 
 might they be communicated to the user?_
 
+I imagine the program can be run on the terminal. At first, the user will be 
+able to import a picture they already own. This is the part where they can call
+the "generate" method, passing on the filename of the image as a parameter. 
+The user might find it inconvenient that they can't see the final product of 
+their coloring page on the terminal since this program is not interactive that way.
+
 
 ## Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy for the user to generate coloring pages with pictures they 
+already own. They willalso be able to specify their own arbitrary settings such as removing all colorsfrom the original pictures except purple. However, it would be impossible or very difficult for the user to take a peak on the final product before the program ends. In other words, they wouldn't be able to interact with the final 
+product in real-time.
+
 
 ## Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well 
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
 
+I had no idea this existed before I just did some research, there's a website 
+that allows users to convert photos into coloring pages, 
+https://www.reallycolor.com/. I want my DSL to essentially do the same thing, 
+but also provide users with different settings such as removing all colorings 
+except a certain color.
+Also, I find it very inconvenient that this website doesn't offer the service 
+for free.
 
 ## Suitability
 _What percentage of your time do you think will be spent directly engaging in
 the **language** aspects of this project (e.g., making language design
 decisions), as opposed to "systems" aspects of the project (e.g., implementing a
 complicated semantics that doesn't require a lot of language design)?_
 
+Developing language aspects of the project would be time-consuming since 
+manipulating colors and other things of photos can be hard. But I don't imagine 
+this project needing to integrate with any APIs. And the system part would be 
```

Comment:
I can see APIs being quite useful actually. Here's an open-source DSL I found that has cool features and shows a picture for the before and the after the feature is applied - one feature that may be particularly useful is the replaceColor that allows a target Color to be replaced by another one: http://imageprocessor.org/imageprocessor/imagefactory/replacecolor/